cmd/juju/firewall/setrule.go

```diff
@@ -52,76 +56,95 @@ func NewSetFirewallRuleCommand() cmd.Command {
 	})
 }
 
 // SetFlags implements cmd.Command.
 func (c *setFirewallRuleCommand) SetFlags(f *gnuflag.FlagSet) {
-	f.StringVar(&c.whitelistValue, "whitelist", "", "list of subnets to whitelist")
+	f.StringVar(&c.allowlist, "allowlist", "", "list of subnets to allowlist")
+	f.StringVar(&c.whitelist, "whitelist", "", "")
 }
 
 // Init implements cmd.Command.
 func (c *setFirewallRuleCommand) Init(args []string) (err error) {
 	if len(args) == 1 {
-		c.service = args[0]
-		if c.whitelistValue == "" {
-			return errors.New("no whitelist subnets specified")
+		c.service = firewall.WellKnownServiceType(args[0])
+		if c.allowlist == "" && c.whitelist == "" {
+			return errors.New("no allowlist subnets specified")
+		}
+		if c.allowlist != "" && c.whitelist != "" {
+			return errors.New("cannot specify both whitelist and allowlist")
 		}
-		if err := c.parseCIDRs(&c.whiteList, c.whitelistValue); err != nil {
-			return errors.Annotate(err, "invalid white-list subnet")
+		if args[0] != "ssh" {
+			return errors.NotSupportedf("service %q", args[0])
+		}
+		if err := c.validateCIDRS(c.allowlist + c.whitelist); err != nil {
+			return errors.Trace(err)
 		}
-		return nil
 	}
 	if len(args) == 0 {
 		return errors.New("no well known service specified")
 	}
 	return cmd.CheckEmpty(args[1:])
 }
 
-func (c *setFirewallRuleCommand) parseCIDRs(cidrs *[]string, value string) error {
-	if value == "" {
-		return nil
-	}
+func (c *setFirewallRuleCommand) validateCIDRS(value string) error {
 	rawValues := strings.Split(value, ",")
 	for _, cidrStr := range rawValues {
 		cidrStr = strings.TrimSpace(cidrStr)
 		if _, _, err := net.ParseCIDR(cidrStr); err != nil {
-			return err
+			return errors.NotValidf(cidrStr)
 		}
-		*cidrs = append(*cidrs, cidrStr)
 	}
 	return nil
 }
 
 // SetFirewallRuleAPI defines the API methods that the set firewall rules command uses.
 type SetFirewallRuleAPI interface {
 	Close() error
-	SetFirewallRule(service string, whiteListCidrs []string) error
+	ModelSet(config map[string]interface{}) error
 }
 
-func (c *setFirewallRuleCommand) Run(_ *cmd.Context) error {
+var deprecationWarning = `
+Firewall rules have been moved to model-config settings "ssh-allow" and
+"saas-ingress-allow". This command is deprecated in favour of
+reading/writing directly to these settings.
+`[1:]
+
+func (c *setFirewallRuleCommand) Run(ctx *cmd.Context) error {
+	if c.whitelist != "" {
+		c.allowlist = c.whitelist
+		ctx.Warningf("--whitelist is deprecated in favour of --allowlist")
+	}
+	ctx.Warningf(deprecationWarning)
+
 	client, err := c.newAPIFunc()
 	if err != nil {
 		return err
 	}
 	defer client.Close()
-	err = client.SetFirewallRule(c.service, c.whiteList)
+
+	switch c.service {
+	case firewall.SSHRule:
+		err = client.ModelSet(map[string]interface{}{config.SSHAllowKey: c.allowlist})
+	case firewall.JujuApplicationOfferRule:
+		err = client.ModelSet(map[string]interface{}{config.SaasIngressAllowKey: c.allowlist})
```

Comment:
What happens if you set SSH first and then SAASIngress? 

We should merge here before setting, so you read the in the client Model and then overwrite the ruleset.